services/web-app/components/catalog/catalog.js

```diff
@@ -134,7 +135,7 @@ function Catalog({ collection, data, type, filter: defaultFilter = {}, glob = {}
     setFilteredAssets(
       assets
         .sort(assetSortComparator(sort))
-        .filter((asset) => assetIsInFilter(asset, filter))
+        .filter((asset) => assetIsInFilter(asset, getCleanFilter(filter)))
```

Comment:
Can we remove this `getCleanFilter` and the 3 below it, and instead, run it in the `handleFilter()` callback because that's the only place that the filter is set? Have that return `setFilter(getCleanFilter(updatedFilter))`?